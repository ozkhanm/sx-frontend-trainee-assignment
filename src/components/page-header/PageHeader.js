import React, {useState} from "react";
import {connect} from "react-redux";
import {Link} from "react-router-dom";

import {ActionCreator} from "../../reducer/action-creator";
import {Operation} from "../../reducer/reducer";

const PageHeader = (props) => {
    const {changeActiveArticleLoadingStatus, dropActiveArticle,
        getArticles, changeCommentsLoadingStatus, activeArticleId, getActiveArticle,
        isCommentLoaded, refreshStatus, changeRefreshStatus, page} = props;
    const [refreshButtonToggled, setRefreshButtonToggleStatus] = useState(false);

    setTimeout(() => {
        setRefreshButtonToggleStatus(false);
        changeRefreshStatus(false);
    }, 3000);

    const renderHeaderControls = () => {
        if (page === `MAIN_PAGE`) {
            return (
                <div className="header-controls">
                    <a className={`header-controls-button header-controls-refresh-button ${(refreshButtonToggled || refreshStatus) ? `header-controls-refresh-button-animation` : ``}`} onClick={() => {
                        setRefreshButtonToggleStatus(true);
                        getArticles();
                    }}/>
                </div>
            );
        } else {
            return (
                <div className="header-controls">
                    <a className={`header-controls-button header-controls-refresh-button ${((refreshButtonToggled && !isCommentLoaded) || refreshStatus) ? `header-controls-refresh-button-animation` : ``}`} onClick={() => {
                        setRefreshButtonToggleStatus(true);
                        changeCommentsLoadingStatus(false);
                        getActiveArticle(activeArticleId);
                    }}/>
                    <Link className="header-controls-button header-controls-return-button" to={`/`} onClick={() => {
                        dropActiveArticle();
                        changeActiveArticleLoadingStatus(false);
                    }}/>
                </div>
            );
        }
    };

    return (
        <React.Fragment>
            <div>
                {renderHeaderControls()}
                <p className="header">Hacker News</p>
            </div>
        </React.Fragment>
    );
};

const mapStateToProps = (state) => ({
    activeArticleId: state.activeArticleId,
    isCommentLoaded: state.isCommentLoaded,
    refreshStatus: state.refreshStatus
});

const mapDispatchToProps = (dispatch) => ({
    changeActiveArticleLoadingStatus(status) {
        dispatch(ActionCreator.changeActiveArticleLoadingStatus(status));
    },
    dropActiveArticle() {
        dispatch(ActionCreator.dropActiveArticle());
    },
    getArticles() {
        dispatch(Operation.getArticles());
    },
    changeCommentsLoadingStatus(status) {
        dispatch(ActionCreator.changeCommentsLoadingStatus(status));
    },
    getActiveArticle(articleId) {
        dispatch(Operation.getActiveArticle(articleId));
    },
    changeRefreshStatus(status) {
        dispatch(ActionCreator.changeRefreshStatus(status));
    }
});

export {PageHeader};
export default connect(mapStateToProps, mapDispatchToProps)(PageHeader);