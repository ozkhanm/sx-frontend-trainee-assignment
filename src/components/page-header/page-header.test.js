import React from "react";
import renderer from "react-test-renderer";
import {Router} from "react-router-dom";

import {PageHeader} from "./PageHeader";

import history from "../../history";

it(`Page header renders correctly on article page screen`, () => {
    const tree = renderer
        .create(
            <Router history={history}>
                <PageHeader page={`ARTICLE_PAGE`}/>
            </Router>)
        .toJSON();
  
    expect(tree).toMatchSnapshot();
});

it(`Page header renders correctly on main page screen`, () => {
    const tree = renderer
        .create(
            <Router history={history}>
                <PageHeader page={`MAIN_PAGE`}/>
            </Router>)
        .toJSON();
  
    expect(tree).toMatchSnapshot();
});