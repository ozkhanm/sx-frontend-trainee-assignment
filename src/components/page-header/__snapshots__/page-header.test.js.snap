// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page header renders correctly on article page screen 1`] = `
<div>
  <div
    className="header-controls"
  >
    <a
      className="header-controls-button header-controls-refresh-button "
      onClick={[Function]}
    />
    <a
      className="header-controls-button header-controls-return-button"
      href="/"
      onClick={[Function]}
    />
  </div>
  <p
    className="header"
  >
    Hacker News
  </p>
</div>
`;

exports[`Page header renders correctly on main page screen 1`] = `
<div>
  <div
    className="header-controls"
  >
    <a
      className="header-controls-button header-controls-refresh-button "
      onClick={[Function]}
    />
  </div>
  <p
    className="header"
  >
    Hacker News
  </p>
</div>
`;
